### DISCUSSION TITLE
Enabling M48 Z_MIN_PROBE_REPEATABILITY_TEST In Precompiled FW

### DISCUSSION THREAD

**Discussion by Ramrawd** [Jyers/Marlin#87] [Ideas]
I've given a go at compiling my own firmware but I fully admit that it's above my skill level at the moment. 

With that said, would it be possible to enable the M48 command for doing a repeatability test using a BLtouch in the precompiled firmwares? 

More info about the command can be found here: https://marlinfw.org/docs/gcode/M048.html

I'd like to figure out how accurate my probe really is!

Thanks!

**Comment by Schnabulation** [comment]
Just to add: I have enabled this in my own config of this firmware and it works flawlessly...

**Reply by Ramrawd** [reply]
That's great to hear! I'm going to give it another go at compiling my own firmware tonight but I think it'd be great if this were included in the prebuilds. The setting itself shouldn't conflict with any other parameters.

**Reply by darco415** [reply]
I have it enabled too but not sure what to look for, is there some sort of log report?

**Reply by Schnabulation** [reply]
Yes, just send M48 in the terminal (through OctoPrint or Pronterface) and watch the probe go. Afterwards the log will be written into the command line.

**Reply by darco415** [reply]
ok I ran the m48 command...not sure what this means:
Recv: Finished!
Recv: Mean: -0.008800 Min: -0.015 Max: -0.005 Range: 0.010
Recv: Standard Deviation: 0.002768
Recv: 
Recv: X:154.10 Y:122.00 Z:12.67 E:0.00 Count X:12328 Y:9760 Z:5068

**Reply by Schnabulation** [reply]
Then maybe go google it..?

Hint: your probe is fine.

**Reply by darco415** [reply]
> Then maybe go google it..?
> 
> Hint: your probe is fine.

google..lol...you must work at a helpdesk....

**Reply by Schnabulation** [reply]
I mean you asked for the M48 command to be added to the firmware, so you should know how to use it. We are not your personal support...

**Reply by darco415** [reply]
> I mean you asked for the M48 command to be added to the firmware, so you should know how to use it. We are not your personal support...

No the OP did.....my question was just on interpretation.  Are you affiliated with this fw or just a troll?

**Reply by Jyers** [reply]
Okay guys, chill. No pointless arguing in some random discussion.

**Reply by darco415** [reply]
> Okay guys, chill. No pointless arguing in some random discussion.

sorry boss

**Reply by Schnabulation** [reply]
> Okay guys, chill. No pointless arguing in some random discussion.

Sorry, you are right...

There are a bunch of info regarding the BL Touch repeatability test. But basically you wanna look at „standard deviation“ - it means how far off the probe was between the different probings. The lower the better - your value is fine for an original BL Touch.

**Comment by Jyers** [comment]
I don't see any reason not to enable this functionality. I'll add it to the next release :)

**Reply by Ramrawd** [reply]
Excellent! Thank you so much for your time!
